tests/sqlparser_snowflake.rs

```diff
@@ -2783,6 +2783,14 @@ fn test_parentheses_overflow() {
 }
 
 #[test]
+
+fn test_nested_select_with_lateral_flatten() {
```

Comment:
Since the functionality doesnt seem to be related to the lateral flatten, can we rename the test to e.g. `test_projection_with_nested_trailing_commas`? And we can include a few more cases (e.g. doubly nested, multiple set expressions that have trailing commas)?

Reply:
can you elaborate on "multiple set expressions that have trailing commas" ?

Reply:
Ah I was thinking something like this where the `trailing_commas` functionality is in use with the `supports_projection_trailing_commas` flag
```rust
SELECT a, b, FROM c, (SELECT d, e, FROM f, LATERAL FLATTEN(input => events))
```